### PR TITLE
tests/gnrc_lorawan: add initial tests

### PR DESCRIPTION
**Pull request by jia200x**
### Contribution description
This PR adds some initial ~unit~test for GNRC LoRaWAN  (EDIT: as suggested by @miri64 , they are now regular tests instead of unittests) . I plan to keep extending these tests in order to improve the overall coverage.

I also piggyback a dependency resolution fix that was discovered while working on the tests.
- `netdev_layered` is not used anymore
- `gnrc_lorawan` actually doesn't depend on any GNRC core module. This is important, because the tests needs to implement some callback functions that are already implemented in `gnrc_netif_lorawan`).

### Testing procedure

`make -C tests/gnrc_lorawan all test` should pass.

### Issues/PRs references
None so far.

### CODE REVIEW

File: tests/unittests/tests-gnrc_lorawan/Makefile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+DEVELHELP = 1
```
Comment (fjmolinas) [inline tests/unittests/tests-gnrc_lorawan/Makefile:1]: Is this needed?

File: tests/unittests/tests-gnrc_lorawan/tests-gnrc_lorawan.c [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+#include "net/gnrc/lorawan.h"
+#include "net/lorawan/hdr.h"
+
+static uint8_t nwkskey[] = {0x44, 0x02, 0x42, 0x41, 0xed, 0x4c, 0xe9, 0xa6, 0x8c, 0x6a, 0x8b, 0xc0, 0x55, 0x23, 0x3f, 0xd3};
+static uint8_t lorawan_packet_no_mic[] = {0x40, 0xF1, 0x7D, 0xBE, 0x49, 0x00, 0x02, 0x00, 0x01, 0x95, 0x43, 0x78, 0x76};
+static uint8_t lorawan_packet_wrong[] = {0x40, 0xF1, 0x7D, 0x00, 0x49, 0x00, 0x02, 0xAA, 0x01, 0x95, 0x43, 0x78, 0x76};
+static uint8_t mic[] = {0x2B, 0x11, 0xFF, 0x0D};
+static le_uint32_t dev_addr ={
+   .u8 =  {0xF1, 0x7D, 0xBE, 0x49}
+};
```
Comment (fjmolinas) [inline tests/unittests/tests-gnrc_lorawan/tests-gnrc_lorawan.c:30]: Seems this file could be uncrustified

**Review by fjmolinas — COMMENTED** [review]
Looks good, some minor comments..

**Comment by jia200x** [comment]
done!

Unfortunately I will need to add symbols to prevent `gnrc_netif_lorawan` defining the callbacks of GNRC LoRaWAN. Since these callbacks are `extern void` functions, the implementations of `gnrc_netif_lorawan` and the test collide :/. I'm currently on holidays, so I won't be able to fix it during these weeks.

**Comment by jia200x** [comment]
updated to regular tests, as proposed by @miri64

**Comment by jia200x** [comment]
green :)

**Review by fjmolinas — APPROVED** [review]
ACK on my side, tests are passing changes look good.

**Comment by jia200x** [comment]
thanks for the review!

**Pull request merged**